os_migrate/plugins/module_utils/subnet.py

```diff
@@ -70,6 +70,11 @@ def _find_sdk_res(conn, name_or_id):
 
     @staticmethod
     def _update_sdk_res(conn, name_or_id, sdk_params):
+        # intermittently the subnet update fails when trying to update with a
+        # message that states the ip_version or network_id attributes cannot
+        # be updated.  See #140 for more information.
+        sdk_params.pop("ip_version")
+        sdk_params.pop("network_id")
```

Comment:
Now it complains about `cidr`. Also we should fix the commit message -- this doesn't fix #140 (idempotence), it fixes #142 (making update work). Either one of them fixed is enough to unblock CI, but they are distinct bugs.

Reply:
IMO we have a third issue here.  If I can run the func tests for the PR locally and they pass with no issues, but then fail in CI, then local tests become worthless.

Reply:
These should be the only fields not for update in Neutron for Subnets: https://github.com/openstack/neutron/blob/8c7ca776d8cbe967a8bbe773ab38c361414a7068/neutron/objects/subnet.py#L232